subprojects/docs/src/docs/userguide/running-builds/introduction/gradle_optimizations.adoc

```diff
@@ -0,0 +1,107 @@
+// You may obtain a copy of the License at
+//
+//      https://creativecommons.org/licenses/by-nc-sa/4.0/
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[gradle_optimizations]]
+= Gradle Incremental Builds and Build Caching
+
+Gradle uses two main features to reduce build time, *incremental builds* and *build caching*.
+
+image::gradle-basic-8.png[]
+
+== Incremental builds
+
+An *incremental build* is a build that avoids running tasks whose inputs have not changed since the previous build, making the execution of such tasks unnecessary.
+
+For incremental builds to work, tasks must define their inputs and outputs.
+Gradle will determine whether the input or outputs have changed at build time.
+If they have changed, Gradle will execute the task.
+Otherwise, it will skip execution.
+
+Incremental builds are always enabled so the only way to see it in action is to turn on _verbose mode_.
+With verbose mode, each task state is labeled during a build:
+
+[source]
+----
+$ ./gradlew compileJava
+
+> Task :buildSrc:generateExternalPluginSpecBuilders UP-TO-DATE
+> Task :buildSrc:extractPrecompiledScriptPluginPlugins UP-TO-DATE
+> Task :buildSrc:compilePluginsBlocks UP-TO-DATE
+> Task :buildSrc:generatePrecompiledScriptPluginAccessors UP-TO-DATE
+> Task :buildSrc:generateScriptPluginAdapters UP-TO-DATE
+> Task :buildSrc:compileKotlin UP-TO-DATE
+> Task :buildSrc:compileJava NO-SOURCE
+> Task :buildSrc:compileGroovy NO-SOURCE
+> Task :buildSrc:pluginDescriptors UP-TO-DATE
+> Task :buildSrc:processResources UP-TO-DATE
+> Task :buildSrc:classes UP-TO-DATE
+> Task :buildSrc:jar UP-TO-DATE
+> Task :list:compileJava UP-TO-DATE
+> Task :utilities:compileJava UP-TO-DATE
+> Task :app:compileJava UP-TO-DATE
+
+BUILD SUCCESSFUL in 374ms
+12 actionable tasks: 12 up-to-date
+----
+
+When you run a task that has been previously executed and hasn't changed, then `UP-TO-DATE` is printed next to the task.
+
+== Build caching
+
+Incremental Builds are a great optimization that helps avoid work already done.
+If a developer continuously changes a single file, there is likely no need to rebuild all the other files in the project.
+
+However, what happens when the same developer switches to a new branch created last week?
+The files are rebuilt, even though the developer is building something that has been built before.
+
+This is where a *build cache* is helpful.
+
+The build cache stores previous build results and restores them when needed.
+It prevents the redundant work and cost of regenerating time-consuming and expensive processes.
+
+When the build cache has been used to repopulate the local directory, the tasks are marked as `FROM-CACHE`:
+
+[source]
+----
+$ ./gradlew compileJava
```

Comment:
```suggestion
$ ./gradlew compileJava --build-cache
```

Does it make sense to link to the build cache documentation [page](https://docs.gradle.org/current/userguide/build_cache.html) from here as well?

Reply:
Thank you, Piotr and I removed steps that made it look more like a tutorial than a reference page. This fixed my issue!